source/os_specific/service_layers/oszephyr.c

```diff
@@ -773,11 +772,21 @@ ACPI_PHYSICAL_ADDRESS
 AcpiOsGetRootPointer (
     void)
 {
+	 LOG_DBG ("");
 
-    LOG_DBG ("");
-    return ((ACPI_PHYSICAL_ADDRESS) efi_get_acpi_rsdp ());
-}
+	if(RsdpPhyAdd)
+	{
+		return RsdpPhyAdd;
+	}
 
+#if defined(CONFIG_X86_EFI)
```

Comment:
No you are still bringing CONFIG_X86_EFI from zephyr. Just get a unique function from zephyr to call, as proposed